snakemake/workflow.py

```diff
@@ -1004,27 +1037,31 @@ def conda_list_envs(self):
         return True
 
     def conda_create_envs(self):
+        assert self.dag_settings is not None
         self._prepare_dag(
             forceall=self.dag_settings.forceall,
             ignore_incomplete=True,
             lock_warn_only=False,
         )
         self._build_dag()
 
+        assert self.deployment_settings is not None
         if DeploymentMethod.APPTAINER in self.deployment_settings.deployment_method:
             self.dag.pull_container_imgs()
         self.dag.create_conda_envs()
 
     def conda_cleanup_envs(self):
+        assert self.dag_settings is not None
         self._prepare_dag(
-            forceall=self.dag_settings.forceall,
+            forceall=self.dag_settings.forceall,  # True?
```

Comment:
yes, why not?

Reply:
Sorry for the trailing comment added here, I've deleted it